### PR TITLE
display issues on TAN-Eingabe

### PR DESCRIPTION
**Pull request by freshking**
## Description

Fix layout issues when using a scroll view in the top view controller

## Link to Jira

https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7112

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/Containers/TopBottomContainerViewController.swift [hunk 1]
```diff
@@ -75,71 +79,94 @@ class TopBottomContainerViewController<TopViewController: UIViewController, Bott
...
-			guard let keyboardSize = (notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue else {
-				return
+			.store(in: &keyboardSubscriptions)
+		
+		NotificationCenter.default.ocombine.publisher(for: UIApplication.keyboardDidHideNotification)
+			.sink { [weak self] _ in
+				guard let self = self else {
+					return
+				}
+				self.footerViewHandler?.didHideKeyboard()
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/Containers/TopBottomContainerViewController.swift:158]: I think we can simplify this
```suggestion
				self?.footerViewHandler?.didHideKeyboard()
```

**Review by whiskey — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Containers/TopBottomContainerViewController.swift (hunk 1 above)
Comment (freshking) [reply src/xcode/ENA/ENA/Source/Scenes/Containers/TopBottomContainerViewController.swift]: Done

**Comment by KaiTeuber** [comment]
Did you tested all the cases where we use `TopBottomContainerViewController`? I'm a little afraid it could break some other screens / flows - like last time we changed that one.

**Comment by freshking** [comment]
> Did you tested all the cases where we use `TopBottomContainerViewController`? I'm a little afraid it could break some other screens / flows - like last time we changed that one.

I tested all the screens I could think of including Onboarding. Maybe I missed one somewhere but the UI tests also succeed so I wouldn't worry too much about it.

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Containers/TopBottomContainerViewController.swift [hunk 2]
```diff
@@ -75,71 +79,90 @@ class TopBottomContainerViewController<TopViewController: UIViewController, Bott
...
-
-
-		keyboardDidShownObserver = NotificationCenter.default.addObserver(forName: UIResponder.keyboardDidShowNotification, object: nil, queue: OperationQueue.main) { notification in
-			guard let keyboardSize = (notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue else {
-				return
+			.store(in: &keyboardSubscriptions)
+		
+		NotificationCenter.default.ocombine.publisher(for: UIApplication.keyboardDidHideNotification)
+			.sink { [weak self] _ in
+				self?.footerViewHandler?.didHideKeyboard()
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/Containers/TopBottomContainerViewController.swift:154]: Why not calling this line in the other sink? Both are registered to UIApplication.keyboardDidHideNotification

File: src/xcode/ENA/ENA/Source/Scenes/Containers/TopBottomContainerViewController.swift [hunk 3]
```diff
@@ -204,23 +227,21 @@ class TopBottomContainerViewController<TopViewController: UIViewController, Bott
...
-	private let initialHeight: CGFloat
 
 	private var subscriptions: [AnyCancellable] = []
+	private var keyboardSubscriptions: [AnyCancellable] = []
 	private var bottomViewHeightAnchorConstraint: NSLayoutConstraint!
-	private var bottomViewBottomAnchorConstraint: NSLayoutConstraint!
-
-	private var keyboardDidShownObserver: NSObjectProtocol?
-	private var keyboardDidHideObserver: NSObjectProtocol?
+	private var bottomViewBottomAnchorAnchorConstraint: NSLayoutConstraint!
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/Containers/TopBottomContainerViewController.swift:234]:
```suggestion
	private var bottomViewBottomConstraint: NSLayoutConstraint!
```

File: src/xcode/ENA/ENA/Source/Scenes/FooterViewController/FooterViewController.swift [hunk 4]
```diff
@@ -91,7 +117,9 @@ class FooterViewController: UIViewController {
 	private let primaryButton: ENAButton = ENAButton(type: .custom)
 	private let secondaryButton: ENAButton = ENAButton(type: .custom)
 
-	private var buttonConstraints = [NSLayoutConstraint]()
+	private var buttonsStackView: UIStackView!
+	private var primaryButtonHeightAnchor: NSLayoutConstraint!
+	private var secondaryButtonHeightAnchor: NSLayoutConstraint!
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/FooterViewController/FooterViewController.swift:122]:
```suggestion
	private var secondaryButtonHeightConstraint: NSLayoutConstraint!
```

File: src/xcode/ENA/ENA/Source/Scenes/FooterViewController/FooterViewController.swift [hunk 5]
```diff
@@ -119,13 +147,22 @@ class FooterViewController: UIViewController {
 		subscription.forEach { $0.cancel() }
 		subscription.removeAll()
 		
-		NSLayoutConstraint.deactivate(buttonConstraints)
-		buttonConstraints.removeAll()
+		primaryButton.isHidden = true
+		secondaryButton.isHidden = true
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/FooterViewController/FooterViewController.swift:151]: These two lines are confusing. Could you please add a comment, why this is happening.

**Review by ArturFriesen — DISMISSED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/FooterViewController/FooterViewController.swift (hunk 5 above)
Comment (freshking) [reply src/xcode/ENA/ENA/Source/Scenes/FooterViewController/FooterViewController.swift:152]: Done

File: src/xcode/ENA/ENA/Source/Scenes/FooterViewController/FooterViewController.swift (hunk 4 above)
Comment (freshking) [reply src/xcode/ENA/ENA/Source/Scenes/FooterViewController/FooterViewController.swift]: Done

File: src/xcode/ENA/ENA/Source/Scenes/FooterViewController/FooterViewController.swift [hunk 6]
```diff
@@ -91,7 +117,9 @@ class FooterViewController: UIViewController {
 	private let primaryButton: ENAButton = ENAButton(type: .custom)
 	private let secondaryButton: ENAButton = ENAButton(type: .custom)
 
-	private var buttonConstraints = [NSLayoutConstraint]()
+	private var buttonsStackView: UIStackView!
+	private var primaryButtonHeightAnchor: NSLayoutConstraint!
```
[the remark this answers is not in the document]
Comment (freshking) [reply src/xcode/ENA/ENA/Source/Scenes/FooterViewController/FooterViewController.swift]: Done

File: src/xcode/ENA/ENA/Source/Scenes/Containers/TopBottomContainerViewController.swift (hunk 3 above)
Comment (freshking) [reply src/xcode/ENA/ENA/Source/Scenes/Containers/TopBottomContainerViewController.swift]: Done

File: src/xcode/ENA/ENA/Source/Scenes/Containers/TopBottomContainerViewController.swift (hunk 2 above)
Comment (freshking) [reply src/xcode/ENA/ENA/Source/Scenes/Containers/TopBottomContainerViewController.swift:154]: There was a mistake. The other sink was supposed to be on `keyboardDidShowNotification`. I fixed it.

**Review by ArturFriesen — APPROVED** [review]

**Pull request merged**